commit dc4994fcc93fd65e680ab70084cae0c46af738b5 (refs/changes/49/4149/2)
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2018-06-07T07:36:28-07:00 (1 year, 4 months ago)
    
    MANATEE-397 `manatee-adm history` fails with `TypeError: Cannot read property 'zoneId' of null` (again)
    Reviewed by: Jared Morrow <jm@joyent.com>

diff --git a/lib/adm.js b/lib/adm.js
index 92b2636..43afc5d 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -1927,6 +1927,11 @@ function annotateHistoryNode(evt, last)
     }
 
     if (nst.generation > lst.generation) {
+        if (lst.sync === null && lst.generation == 0 &&
+            lst.oneNodeWriteMode !== true) {
+            return (sprintf('upgrade to 2.0, sync "%s" added',
+                nst.sync.zoneId.substr(0, 8)));
+        }
         if (nst.sync.zoneId == lst.sync.zoneId) {
             return ('error: gen number changed, but ' +
                 'primary and sync did not');
